CMakeLists.txt

```diff
@@ -2003,6 +2003,11 @@ if(CONFIG_MINIMAL_LIBCPP)
   zephyr_compile_options($<$<COMPILE_LANGUAGE:CXX>:$<TARGET_PROPERTY:compiler-cpp,nostdincxx>>)
 endif()
 
+if(CONFIG_LLVM_PROFILE)
+  zephyr_compile_options(-fprofile-instr-generate -fcoverage-mapping)
```

Comment:
This should probably be commented by @tejlmand (meaning, if he proposes something else, ignore what I say here), but would it not be nicer if instead of doing this, 
We'd have the definition of these two
`$<TARGET_PROPERTY:compiler,coverage>` and `$<TARGET_PROPERTY:linker,coverage>`
be based on the kconfig choice?

Note: `compiler,coverage` is defined in:
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/compiler/gcc/compiler_flags.cmake#L156-L157
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/compiler/clang/compiler_flags.cmake#L17-L18
And `linker,coverage` in:
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/linker/ld/clang/linker_flags.cmake
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/linker/ld/gcc/linker_flags.cmake#L7

Reply:
I think it makes a lot of sense to use the `coverage` property for this.

That will keep the code in common CMakeLists.txt generic and thus allow other toolchains decide / implement the flags to use which provides the desired functionality.